packages/landingpage-new/components/about/activity.tsx

```diff
@@ -24,96 +25,116 @@ ChartJS.register(
+): [string[], number[]] {
+  const graphActivities: GraphActivity[] = [];
 
-      const yearAction = graphActivities.find((action) => action.month === currentMonth && action.year === currentYear);
-      if (yearAction) {
-        yearAction.contributions = yearAction.contributions + activity.total;
-      } else if (currentMonth && currentYear) {
-        graphActivities.push({
-          contributions: activity.total,
-          month: currentMonth,
-          year: currentYear
+  githubActivities.forEach((activity) => {
+    const currentDate = activity.date
+      ? activity.date.toLocaleString('en-US', {
+          month: 'short',
+          year: '2-digit',
         })
-      }
-    });
-    return graphActivities;
+      : undefined;
+
+    const yearAction = graphActivities.find(
+      (action) => action.date === currentDate
+    );
+    if (yearAction) {
+      yearAction.contributions = yearAction.contributions + activity.total;
+    } else if (currentDate) {
+      graphActivities.push({
+        contributions: activity.total,
+        date: currentDate,
+      });
+    }
+  });
+  return [
+    graphActivities.map((activity) => activity.date),
+    graphActivities.map((activity) => activity.contributions),
+  ];
 }
 
 async function getGithubActivities(): Promise<GithubActivity[]> {
-    return await fetch(
-        GITHUB_REPO_URL + '/stats/commit_activity'
-    ).then((resp) => resp.json());
+  return await fetch(GITHUB_REPO_URL + '/stats/commit_activity').then((resp) =>
+    resp.json()
+  );
 }
 
+const graphOptions = {
+  responsive: true,
+  plugins: {
+    legend: {
+      position: 'top' as const,
+    },
+  },
+};
+
 function Activity() {
-    const [graphActivities, setGraphActivities] = useState([] as GraphActivity[]);
+  const [labels, setLabels] = useState([] as string[]);
+  const [contributions, setContributions] = useState([] as number[]);
+  const [graphData, setGraphData] = useState({
+    datasets: [],
+    labels: [],
+  } as ChartData<'line', number[], unknown>);
 
-    useEffect(() => {
-      getGithubActivities().then(githubActivities => setGraphActivities(generateGraphActivities(githubActivities)))
-    }, [])
+  useEffect(() => {
+    getGithubActivities().then((githubActivities) => {
+      githubActivities = fillDate(githubActivities);
+      const [newLabels, newContributions] = generateGraphData(githubActivities);
+      setLabels(newLabels);
+      setContributions(newContributions);
+    });
+  }, []);
 
-    const graphData = {
-      labels: graphActivities.map((graphActivity) => graphActivity.month + ' ' + graphActivity.year),
+  useEffect(() => {
+    setGraphData({
+      labels,
       datasets: [
         {
           fill: true,
           label: 'contributions',
-          data: graphActivities.map(graphActivity => graphActivity.contributions),
+          data: contributions,
           borderColor: '#4F46FF',
           backgroundColor: '#BECCFF50',
         },
       ],
```

Comment:
Can we move this to the `useEffect` hook above? That way, the `label` and `contributions`  will be unnecessary and all of our data lies in `graphData`